### DISCUSSION TITLE
Extending fileMatch in gitlabci/gitlabci-include doesn't seem to work

### DISCUSSION THREAD

**Discussion by syphernl** [renovatebot/renovate#25629] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.50.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

In a few repo's we generate the GitLab CI configuration. For this we use `.gitlab-ci.yml.tpl` as file which contains a template that gets rendered.
The template file iteself is basically "just" a GitLab CI config but with some templates/loops etc added in. But it also contains `include:`  which we need to have updated alongside changes in the `.gitlab-ci.yml`.

By default the `fileMatch` for both managers is:

```yaml
{
  "fileMatch": [
    "\\.gitlab-ci\\.ya?ml$"
  ]
}
```

I have configured our Renovate to:

```yaml
  gitlabci: {
    fileMatch: ["\\.gitlab-ci\\.ya?ml(\\.tpl)?$"]
  },
  'gitlabci-include': {
    fileMatch: ["\\.gitlab-ci\\.ya?ml(\\.tpl)?$"]
  },
```

It seems that Renovate is picking up on the additional fileMatches:

```json
{"name":"renovate","hostname":"laptop","pid":104167,"level":20,"logContext":"x","repository":"PROJECT/infra","msg":"Using file match: \\.gitlab-ci\\.ya?ml$ for manager gitlabci","time":"2023-11-07T09:36:01.786Z","v":0}
{"name":"renovate","hostname":"laptop","pid":104167,"level":20,"logContext":"x","repository":"PROJECT/infra","msg":"Using file match: \\.gitlab-ci\\.ya?ml(\\.tpl)?$ for manager gitlabci","time":"2023-11-07T09:36:01.786Z","v":0}
{"name":"renovate","hostname":"laptop","pid":104167,"level":20,"logContext":"x","repository":"PROJECT/infra","msg":"Using file match: \\.gitlab-ci\\.ya?ml$ for manager gitlabci-include","time":"2023-11-07T09:36:01.787Z","v":0}
{"name":"renovate","hostname":"laptop","pid":104167,"level":20,"logContext":"x","repository":"PROJECT/infra","msg":"Using file match: \\.gitlab-ci\\.ya?ml(\\.tpl)?$ for manager gitlabci-include","time":"2023-11-07T09:36:01.787Z","v":0}
```

It however doesn't seem to work as it only finds `.gitlab-ci.yml` and not `.gitlab-ci.yml.tpl` which is also in the repo:

```json
{"name":"renovate","hostname":"laptop","pid":106628,"level":20,"logContext":"x","repository":"PROJECT/infra","msg":"Matched 1 file(s) for manager gitlabci: .gitlab-ci.yml","time":"2023-11-07T09:44:04.499Z","v":0}
{"name":"renovate","hostname":"laptop","pid":106628,"level":20,"logContext":"x","repository":"PROJECT/infra","msg":"Matched 1 file(s) for manager gitlabci-include: .gitlab-ci.yml","time":"2023-11-07T09:44:04.499Z","v":0}
```

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

Placeholder value, please select the correct response from the dropdown

**Comment by rarkins** [comment]
Are the extra files _valid YAML_ ?

**Reply by syphernl** [reply]
I made a small config change:

```json
  gitlabci: {
    // fileMatch: ["\\.gitlab-ci\\.ya?ml(\\.tpl)?$"]
    fileMatch: ["\\.gitlab-ci\\.ya?ml\\.tpl$"]
  },
  'gitlabci-include': {
    // fileMatch: ["\\.gitlab-ci\\.ya?ml(\\.tpl)?$"]
    fileMatch: ["\\.gitlab-ci\\.ya?ml\\.tpl$"]
  },
```
With this I see this additional file does get picked up but indeed the YAML is the problem here as it throws a YAMLException..

**Comment by syphernl** [comment, marked as answer]
I fixed it using this config:

```js
    {
      description: 'Update includes in GitLab CI Templates',
      fileMatch: ["\\.gitlab-ci\\.ya?ml\\.tpl$"],
      matchStrings:
      [
        "\\s*-\\s+project:\\s*\['\"\](?<depName>[^']+)\['\"\]\\s+ref:\\s+(?<currentValue>[^#\\n]+)"
      ],
      datasourceTemplate: 'gitlab-tags',
      versioningTemplate: 'semver',
      depTypeTemplate: 'repository',
      registryUrlTemplate: "https://gitlab.example.com",
    },
```

**Reply by JMora77** [reply]
I set this as renovate.json and got an error with currentValue that was fixed by removing the "+" signs
